deploy-scheduled-ecs/deploy.sh

```diff
@@ -36,19 +36,26 @@ if (echo "${newcontainers}" | jq '.[0] | .secrets' | grep '^null$'); then
   exit 1
 fi
 
+# build register-task-definition command options in array
+reg_options=( --family "${ECS_TASK_DEF}" )
+reg_options+=( --task-role-arn "$(echo "${taskdefinition}" | jq -r '.taskDefinition.taskRoleArn')" )
+reg_options+=( --execution-role-arn "$(echo "${taskdefinition}" | jq -r '.taskDefinition.executionRoleArn')" )
+reg_options+=( --network-mode "$(echo "${taskdefinition}" | jq -r '.taskDefinition.networkMode')" )
+reg_options+=( --container-definitions "${newcontainers}" )
+reg_options+=( --volumes "$(echo "${taskdefinition}" | jq '.taskDefinition.volumes')" )
+reg_options+=( --placement-constraints "$(echo "${taskdefinition}" | jq '.taskDefinition.placementConstraints')" )
+reg_options+=( --requires-compatibilities "$(echo "${taskdefinition}" | jq -r '.taskDefinition.requiresCompatibilities')" )
+reg_options+=( --cpu "$(echo "${taskdefinition}" | jq -r '.taskDefinition.cpu')" )
+reg_options+=( --memory "$(echo "${taskdefinition}" | jq -r '.taskDefinition.memory')" )
+# include --ephemeral-storage option if available in template task definition
+template_ephemeral_storage="$(echo "${taskdefinition}" | jq -r '.taskDefinition.ephemeralStorage')"
+if [ "$template_ephemeral_storage" != null ]; then
+    reg_options+=( --ephemeral-storage "${template_ephemeral_storage}")
+fi
+
 # this task uses fargate, so publish the new task definition
 echo "Publishing new Task Definition."
-aws ecs register-task-definition \
---family "${ECS_TASK_DEF}" \
---task-role-arn "$(echo "${taskdefinition}" | jq -r '.taskDefinition.taskRoleArn')" \
---execution-role-arn "$(echo "${taskdefinition}" | jq -r '.taskDefinition.executionRoleArn')" \
---network-mode "$(echo "${taskdefinition}" | jq -r '.taskDefinition.networkMode')" \
---container-definitions "${newcontainers}" \
---volumes "$(echo "${taskdefinition}" | jq '.taskDefinition.volumes')" \
---placement-constraints "$(echo "${taskdefinition}" | jq '.taskDefinition.placementConstraints')" \
---requires-compatibilities "$(echo "${taskdefinition}" | jq -r '.taskDefinition.requiresCompatibilities')" \
---cpu "$(echo "${taskdefinition}" | jq -r '.taskDefinition.cpu')" \
---memory "$(echo "${taskdefinition}" | jq -r '.taskDefinition.memory')"
+aws ecs register-task-definition "${reg_options[@]}"
```

Comment:
question: are these blocks in the two different files different in any particular way? could we refactor it into a bash function that's shared between them?

Reply:
They should be the same between the two files, I can work on moving them into a function

Reply:
Sorry, I could have been more clear. Since they're the same, can they be a single function that's shared between the two scripts?

Reply:
I put the same function in each file. Did you mean to have one function in a separate file and source that file into both deploy scrips?

Reply:
The latter; the code should only exist once.

Reply:
Maybe a new `shared` or `src` folder at the top level? There's no technical reason that the `.sh` file needs to live in the same folder as the `action.yml` file.